utils/redis_storage.ts

```diff
@@ -0,0 +1,207 @@
+import * as E from "fp-ts/lib/Either";
+import { pipe } from "fp-ts/lib/function";
+import * as O from "fp-ts/lib/Option";
+import * as TE from "fp-ts/lib/TaskEither";
+import { RedisClient } from "redis";
+
+/**
+ * Parse a Redis given string reply.
+ *
+ * @see https://redis.io/topics/protocol#simple-string-reply.
+ */
+const givenStringReply = (
+  err: Error | null,
+  reply: string | undefined,
+  message: string
+): E.Either<Error, boolean> => {
+  if (err) {
+    return E.left<Error, boolean>(err);
+  }
+
+  return E.right<Error, boolean>(reply === message);
+};
+
+/**
+ * Parse a Redis single string reply.
+ *
+ * @see https://redis.io/topics/protocol#simple-string-reply.
+ */
+const singleStringReply = (
+  err: Error | null,
+  reply: "OK" | undefined
+): E.Either<Error, boolean> => {
+  if (err) {
+    return E.left<Error, boolean>(err);
+  }
+
+  return E.right<Error, boolean>(reply === "OK");
+};
+
+/**
+ * Parse a Redis single string reply.
+ *
+ * @see https://redis.io/topics/protocol#simple-string-reply.
+ */
+const singleValueReply = (
+  err: Error | null,
+  reply: string | null
+): E.Either<Error, O.Option<string>> => {
+  if (err) {
+    return E.left<Error, O.Option<string>>(err);
+  }
+  return E.right<Error, O.Option<string>>(O.fromNullable(reply));
+};
+
+/**
+ * Parse a Redis integer reply.
+ *
+ * @see https://redis.io/topics/protocol#integer-reply
+ */
+const integerReply = (
+  err: Error | null,
+  reply: unknown,
+  expectedReply?: number
+): E.Either<Error, boolean> => {
+  if (err) {
+    return E.left<Error, boolean>(err);
+  }
+  if (expectedReply !== undefined && expectedReply !== reply) {
+    return E.right<Error, boolean>(false);
+  }
+  return E.right<Error, boolean>(typeof reply === "number");
+};
+
+const falsyResponseToError = (
+  response: E.Either<Error, boolean>,
+  error: Error
+): E.Either<Error, true> => {
+  if (E.isLeft(response)) {
+    return response;
+  } else {
+    if (response.right) {
+      return E.right(true);
+    }
+    return E.left(error);
+  }
+};
```

Comment:
More functional implementation
```suggestion
const isTrue = (b: boolean): b is true => b === true;

const falsyResponseToError = (
  response: E.Either<Error, boolean>,
  error: Error
): E.Either<Error, true> =>
  pipe(
    response,
    E.filterOrElse(isTrue, () => error)
```